CONVENTIONS.md

```diff
@@ -85,3 +85,86 @@ metal3-io](https://github.com/metal3-io/metal3-docs/blob/master/design/bare-meta
+tolerate the following taint:
+
+* `node-role.kubernetes.io/master`
+
+For example:
+
+```yaml
+spec:
+  template:
+    spec:
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        operator: Exists
+        effect: NoSchedule
+```
+
+Tolerating this taint should suffice for the vast majority of core OpenShift
+operators.  In exceptional cases, an operator may tolerate one or more of the
+following taints if doing so is necessary to form a functional Kubernetes node:
+
+* `node.kubernetes.io/disk-pressure`
+* `node.kubernetes.io/memory-pressure`
+* `node.kubernetes.io/network-unavailable`
+* `node.kubernetes.io/not-ready`
+* `node.kubernetes.io/pid-pressure`
+* `node.kubernetes.io/unreachable`
+
+Operators should not specify tolerations in their manifests for any of the taints in
+the above list without an explicit and credible justification.
+
+When an operator configures its operand, the operator likewise may specify
+tolerations for the aforementioned taints but should do so only as necessary and only
+with explicit justification.
+
+Note that the DefaultTolerationSeconds and PodTolerationRestriction admission plugins
+may add time-bound tolerations to an operator or operand in addition to the
+tolerations that the operator has specified.
+
+If appropriate, a CRD that corresponds to an operand may provide an API to allow
+users to specify a custom list of tolerations for that operand.  For examples, see
+the
+[imagepruners.imageregistry.operator.openshift.io/v1](https://github.com/openshift/api/blob/34f54f12813aaed8822bb5bc56e97cbbfa92171d/imageregistry/v1/types_imagepruner.go#L67-L69),
+[configs.imageregistry.operator.openshift.io/v1](https://github.com/openshift/api/blob/34f54f12813aaed8822bb5bc56e97cbbfa92171d/imageregistry/v1/types.go#L82-L84),
+[builds.config.openshift.io/v1](https://github.com/openshift/api/blob/34f54f12813aaed8822bb5bc56e97cbbfa92171d/config/v1/types_build.go#L96-L99),
+and
+[ingresscontrollers.operator.openshift.io/v1](https://github.com/openshift/api/blob/34f54f12813aaed8822bb5bc56e97cbbfa92171d/operator/v1/types_ingress.go#L183-L191)
+APIs.
+
+In exceptional cases, an operand may tolerate all taints:
+
+* if the operand is required to form a functional Kubernetes node, or
+* if the operand is required to support workloads sourced from an internal or external registry that core components depend upon,
+
+then the operand should tolerate all taints:
+
+```yaml
+spec:
+  template:
+    spec:
+      tolerations:
+      - operator: Exists
+```
+
+An example of an operand that matches the first case is kube-proxy, which is required
+for services to work.  An example of an operand that matches the second case is the
+DNS node resolver, which adds an entry to the `/etc/hosts` file on all node hosts so
+that the container runtime is able to resolve the name of the cluster image registry;
+absent this entry in `/etc/hosts`, upgrades could fail to pull images of core
+components.
```

Comment:
I think following what's written here, we need to include the MCO as it's responsible for the overall node lifecycle and thus, the node cannot be properly configured otherwise (kubelet among others) https://bugzilla.redhat.com/show_bug.cgi?id=1780318